src/lib/updating-element.ts

```diff
@@ -484,15 +486,13 @@ export abstract class UpdatingElement extends HTMLElement {
 
   connectedCallback() {
     this._updateState = this._updateState | STATE_HAS_CONNECTED;
-    // Ensure connection triggers an update. Updates cannot complete before
+    // Ensure first connection completes an update. Updates cannot complete before
     // connection and if one is pending connection the `_hasConnectionResolver`
     // will exist. If so, resolve it to complete the update, otherwise
     // requestUpdate.
     if (this._hasConnectedResolver) {
       this._hasConnectedResolver();
       this._hasConnectedResolver = undefined;
-    } else {
-      this.requestUpdate();
```

Comment:
Summary here is we're considering the fact that `requestUpdate` was called _every_ connection a bug, since the intention of this code was only to ensure a "request if not yet requested" one time on boot-up.

If users depend on tree-state in `render()`, they should manually `requestUpdate` in `connectedCallback` to ensure `render` is called when the tree state changes.